consensus/polybft/stake_manager_fuzz_test.go

```diff
@@ -0,0 +1,281 @@
+			5,
+		)
+
+		// insert initial full validator set
+		require.NoError(t, state.StakeStore.insertFullValidatorSet(validatorSetState{
+			Validators: newValidatorStakeMap(validators.getPublicIdentities(initialSetAliases...)),
+		}))
+
+		receipt := &types.Receipt{
+			Logs: []*types.Log{
+				createTestLogForTransferEvent(
+					t,
+					stakeManager.validatorSetContract,
+					validators.getValidator(initialSetAliases[validatorID]).Address(),
+					types.ZeroAddress,
+					stakeValue,
+				),
+			},
+		}
+
+		req := &PostBlockRequest{
+			FullBlock: &types.FullBlock{Block: &types.Block{Header: &types.Header{Number: block}},
+				Receipts: []*types.Receipt{receipt},
+			},
+			Epoch: epoch,
+		}
+		_ = stakeManager.PostBlock(req)
+	})
+}
+
+func FuzzTestStakeManagerUpdateValidatorSet(f *testing.F) {
+	var (
+		aliases = []string{"A", "B", "C", "D", "E"}
+		stakes  = []uint64{10, 10, 10, 10, 10}
+	)
+
+	validators := newTestValidatorsWithAliasesF(f, aliases, stakes)
+	state := newTestStateF(f)
+
+	stakeManager := newStakeManager(
+		hclog.NewNullLogger(),
+		state,
+		nil,
+		nil,
+		wallet.NewEcdsaSigner(validators.getValidator("A").Key()),
+		types.StringToAddress("0x0001"), types.StringToAddress("0x0002"),
+		10,
+	)
+
+	f.Fuzz(func(t *testing.T, input []byte) {
+		tp, err := go_fuzz_utils.NewTypeProvider(input)
+		if err != nil {
+			return
+		}
+		epoch, err := tp.GetUint64()
+		if err != nil {
+			return
+		}
+		x, err := tp.GetUint64()
+		if err != nil {
+			return
+		}
+		if x > uint64(len(aliases)-1) {
+			t.Skip()
+		}
+		votingPower, err := tp.GetInt64()
+		if err != nil {
+			return
+		}
+
+		_, _ = stakeManager.UpdateValidatorSet(epoch, validators.getPublicIdentities(aliases[x:]...))
+
+		fullValidatorSet := validators.getPublicIdentities().Copy()
+		validatorToUpdate := fullValidatorSet[x]
+		validatorToUpdate.VotingPower = big.NewInt(votingPower)
+
+		_, _ = stakeManager.UpdateValidatorSet(epoch, validators.getPublicIdentities())
+	})
+}
+
+func newTestValidatorsF(f *testing.F, validatorsCount int) *testValidators {
+	f.Helper()
+
+	aliases := make([]string, validatorsCount)
+	for i := 0; i < validatorsCount; i++ {
+		aliases[i] = strconv.Itoa(i)
+	}
+
+	return newTestValidatorsWithAliasesF(f, aliases)
+}
+
+func newTestValidatorsWithAliasesF(f *testing.F, aliases []string, votingPowers ...[]uint64) *testValidators {
+	f.Helper()
+
+	validators := map[string]*testValidator{}
+
+	for i, alias := range aliases {
+		votingPower := uint64(1)
+		if len(votingPowers) == 1 {
+			votingPower = votingPowers[0][i]
+		}
+
+		validators[alias] = newTestValidatorF(f, alias, votingPower)
+	}
+
+	return &testValidators{validators: validators}
+}
+
+func newTestValidatorF(f *testing.F, alias string, votingPower uint64) *testValidator {
+	f.Helper()
+
+	return &testValidator{
+		alias:       alias,
+		votingPower: votingPower,
+		account:     generateTestAccountF(f),
+	}
+}
+
+func generateTestAccountF(f *testing.F) *wallet.Account {
```

Comment:
Same here. No need for this function. We already have it in the `helpers_test.go` just change it to use `testing.TB`:

`func generateTestAccount(t testing.TB) *wallet.Account {`